### PR TITLE
V1.0

### PR DESCRIPTION
**Pull request by parkererickson**
Opening V1.0 PR for doc review. Do not merge until v1.0 of pyTigerGraph is being released, targeted for July 7th, 2022.

### CODE REVIEW

File: modules/core-functions/pages/gsql.adoc [hunk 1]
```diff
@@ -21,3 +21,21 @@ time of establishing the connection will be used.
 The output of the statement(s) executed.
 
 
+== installUDF()
+`installUDF(ExprFunctions: str = "", ExprUtil: str = "") -> None`
+
+Install User Defined Function (UDF) to the database.  +
+See the https://docs.tigergraph.com/gsql-ref/current/querying/func/query-user-defined-functions[documentation] for more details. for details on UDFs.
```
Comment (alexthomasTG) [inline modules/core-functions/pages/gsql.adoc]: Extra half of the sentence after the sentence ends

File: modules/core-functions/pages/gsql.adoc [hunk 2]
```diff
@@ -21,3 +21,21 @@ time of establishing the connection will be used.
...
+
+[discrete]
+==== Parameters:
+ExprFunctions (str, optional): 
+Path to the file for ExprFunctions. Defaults to ''.
```
Comment (alexthomasTG) [inline modules/core-functions/pages/gsql.adoc]: Maybe say "Defaults to '' (a blank path)." so users know there was no mistake when writing the docs

**Review by alexthomasTG — APPROVED** [review]
Two small things, looks good!

File: modules/core-functions/pages/gsql.adoc [hunk 3]
```diff
@@ -24,15 +24,16 @@ The output of the statement(s) executed.
 == installUDF()
 `installUDF(ExprFunctions: str = "", ExprUtil: str = "") -> None`
 
-Install User Defined Function (UDF) to the database.  +
+Install User Defined Function (UDF) to the database.
```
Comment (lennessyy) [inline modules/core-functions/pages/gsql.adoc]: we don't need to capitalize initials for user-defined functions

File: modules/core-functions/pages/loading.adoc [hunk 4]
```diff
@@ -12,6 +12,9 @@ Execute a loading job with the referenced file.
 The file will first be uploaded to the TigerGraph server and the value of the appropriate
 FILENAME definition will be updated to point to the freshly uploaded file.
 
+**Note:** The argument `USING HEADER="true"` in the GSQL loading job may not be enough to
```
Comment (lennessyy) [inline modules/core-functions/pages/loading.adoc]: use `NOTE: ` instead of Note:

`NOTE: ` is an admonition in asciidoc and will convert to an actual note box

File: modules/gds/pages/featurizer.adoc [hunk 5]
```diff
@@ -17,20 +17,25 @@ Prints the information for algorithms within the specified category.
...
+`installAlgorithm(query_name: str, query_path: str = None, global_change: bool = False) -> str`
 
 Checks if the query is already installed. 
 If the query is not installed, it installs the query and changes the schema if an attribute needs to be added.
 
 [discrete]
 ==== Parameters:
-* `query_name (str)`: The name of query to be installed.
+query_name (str): 
+The name of query to be installed
```
Comment (alexthomasTG) [inline modules/gds/pages/featurizer.adoc]: The name of the query to be installed

File: modules/gds/pages/featurizer.adoc [hunk 6]
```diff
@@ -17,20 +17,25 @@ Prints the information for algorithms within the specified category.
...
 * `query_path (str)`: If using a custom query, the path to the `.gsql` file that contains the query.
 Note: you must have the `query_name` parameter match the name of the query in the file.
 
+* `global_change (bool)`: False by default. Set to true if you want to run `GLOBAL SCHEMA_CHANGE JOB`. For Algorithms that are not schema free we need to specify this argument.
```
Comment (alexthomasTG) [inline modules/gds/pages/featurizer.adoc]: No need to capitalize "algorithms."

Maybe reword to "This argument is required for algorithms that are not schema-free."

File: modules/gds/pages/featurizer.adoc [hunk 7]
```diff
@@ -49,10 +54,11 @@ feat_name (str):
 An attribute name that needs to be added to the vertex/edge
 * `feat_type (str)`: Type of attribute that needs to be added to the vertex/edge. Only needed if `custom_query` is set to `True`.
 * `custom_query (bool)`: If the query is a custom query. Defaults to False. 
-* `schema_name (list[str])`: List of Vertices/Edges that the attr_name need to added to them.
+* `schema_name`: List of Vertices/Edges that the attr_name need to added to them.
```
Comment (alexthomasTG) [inline modules/gds/pages/featurizer.adoc:55]: List of vertices/edges that need the `attr_name` added to them

File: modules/gds/pages/metrics.adoc [hunk 8]
```diff
@@ -37,10 +37,12 @@ Get accuracy score.
 Accuracy score (float).
 
 
-== Recall
+== BinaryRecall
 
 Recall = stem:[rac{\sum(preds * labels)}{\sum(labels)}]
 
+This metric is for binary classifications, i.e., both preds and labels are arrays of 0's and 1's.
```
Comment (alexthomasTG) [inline modules/gds/pages/metrics.adoc]: Spell out "predictions"

File: modules/core-functions/pages/base.adoc [hunk 9]
```diff
@@ -16,6 +16,8 @@ Initiate a connection object.
 protocol (http:// or https://). If `certPath` is `None` and the protocol is https,
 a self-signed certificate will be used.
 * `graphname`: The default graph for running queries.
+* `gsqlSecret`: The secret key for GSQL. Created in AdminPortal of GraphStudio for the graph you wish to use.
```
Comment (lennessyy) [inline modules/core-functions/pages/base.adoc]: Added a cross-reference for users to learn about what a secret is, since secrets aren't just created in Admin Portal. Also, Admin Portal isn't considered by the application team to be part of GraphStudio. I think in future designs they will make the distinction more pronounced. 

Replace the description for `gsqlSecret` with the following:

> https://docs.tigergraph.com/tigergraph-server/current/user-access/managing-credentials#_secrets[Secret] used to authenticate requests.

File: modules/gds/pages/featurizer.adoc [hunk 10]
```diff
@@ -49,10 +54,11 @@ feat_name (str):
 An attribute name that needs to be added to the vertex/edge
 * `feat_type (str)`: Type of attribute that needs to be added to the vertex/edge. Only needed if `custom_query` is set to `True`.
 * `custom_query (bool)`: If the query is a custom query. Defaults to False. 
-* `schema_name (list[str])`: List of Vertices/Edges that the attr_name need to added to them.
+* `schema_name`: List of Vertices/Edges that the attr_name need to added to them.  
 * `global_schema (bool)`: False by default. Set to true if you want to run `GLOBAL SCHEMA_CHANGE JOB`.
  +
 See the https://docs.tigergraph.com/gsql-ref/current/ddl-and-loading/modifying-a-graph-schema#_global_vs_local_schema_changes[documentation] for more details.
+for more details.
```
Comment (lennessyy) [inline modules/gds/pages/featurizer.adoc]: This seems redundant

File: modules/core-functions/pages/base.adoc [hunk 11]
```diff
@@ -16,7 +16,8 @@ Initiate a connection object.
 protocol (http:// or https://). If `certPath` is `None` and the protocol is https,
 a self-signed certificate will be used.
 * `graphname`: The default graph for running queries.
-* `gsqlSecret`: The secret key for GSQL. Created in AdminPortal of GraphStudio for the graph you wish to use.
+* `gsqlSecret`: The secret key for GSQL.  +
+See the https://docs.tigergraph.com/tigergraph-server/current/user-access/managing-credentials#_secrets.[documentation] for more details.
```
Comment (lennessyy) [inline modules/core-functions/pages/base.adoc]: Let's not say "See the documentation for more details". This description is in the documentation - so it's weird to tell users to see the documentation when they are already reading the documentation. We should just use the name of the page or the name of the section.

See Secrets for more details | See Managing Credentials for more details.

**Review by lennessyy — APPROVED** [review]

File: modules/core-functions/pages/base.adoc (hunk 11 above)
Comment (parkererickson) [reply modules/core-functions/pages/base.adoc]: I agree with this, but the `[documentation] for more details.` is being generated by the doc generation script for all links in the code comments. I am not sure if we can reliably parse the page urls to get something more specific to be generated here.

Comment (lennessyy) [reply modules/core-functions/pages/base.adoc]: Instead of having the description links generated, maybe we can just write them manually? I don't think we really have that many links in the comments, so i feel like it shouldn't be much of a burden. The description links are just texts like other texts in the description after all
